backend/drm/bo_handle_table.c

```diff
@@ -0,0 +1,43 @@
+#include <assert.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <wlr/util/log.h>
+#include "backend/drm/bo_handle_table.h"
+
+static size_t align(size_t val, size_t align) {
+	size_t mask = align - 1;
+	return (val + mask) & ~mask;
+}
+
+void drm_bo_handle_table_finish(struct wlr_drm_bo_handle_table *table) {
+	free(table->nrefs);
+}
+
+bool drm_bo_handle_table_ref(struct wlr_drm_bo_handle_table *table,
+		uint32_t handle) {
+	assert(handle != 0);
+
+	if (handle > table->len) {
+		// Grow linearily, because we don't expect the number of BOs to explode
+		size_t len = align(handle + 1, 512);
```

Comment:
Why 512?

Reply:
Arbitrary number. This happens to be `4096 / 8`, which is what amdgpu is using.

Reply:
Could the value of a memory page size change depending on the arch or bitness?

Reply:
Maybe, but it's not worth the effort. Growing by a reasonable amount of memory should be enough. Using the page table size (as amdgpu does) is micro-optimization IMHO.

Reply:
Sounds good